### ISSUE TITLE
[Need help] run sudo command in unix.execve()

### ISSUE THREAD

**Issue by diyism** [jart/cosmopolitan#623]
```
malcolm@kali:~/ sudo cat /etc/sudoers
......
malcolm ALL = NOPASSWD: /usr/bin/ls

malcolm@kali:~/ sudo /usr/bin/ls
redbean.zip

malcolm@kali:~/ unzip redbean.zip
malcolm@kali:~/ cat test.lua
local unix = require "unix"
cmd=assert(unix.commandv('/usr/bin/sudo'))
unix.execve(cmd, {cmd, '/usr/bin/ls'})
Write('test')

# start redbean web server and visit https://127.0.0.1:8080/test.lua
malcolm@kali:~/ ./redbean.zip
I2022-09-15T00:52:10.018007:tool/net/redbean.c:6695:redbean:1374151] (srvr) listen http://127.0.0.1:8080
..........
sudo: effective uid is not 0, is /usr/bin/sudo on a file system with the 'nosuid' option set or an NFS file system without root privileges?
W2022-09-15T00:52:21.933534:tool/net/redbean.c:1212:redbean:1374151] (stat) 1374228 exited with 1 (2 workers remain)
........
sudo: effective uid is not 0, is /usr/bin/sudo on a file system with the 'nosuid' option set or an NFS file system without root privileges?
W2022-09-15T00:55:10+000542:tool/net/redbean.c:1212:redbean:1374870] (stat) 1374958 exited with 1 (0 workers remain)
```

I can't figure out how to run sudo command, I'm trying to control a USB power relay to swith on some gadgets via visiting my home-hosted redbean website from outside of my home, but I need "sudo" to run the usb serial port command, and I don't want to run redbean with root user, any hint? Thanks.

**Comment by jart** [comment]
execve() replaces the current process with the spawned command. Your Write() call never executes because the process is gone at that point. You need to call execve() from a forked child process. See the following example to see how to do it with redbean. https://github.com/jart/cosmopolitan/blob/master/tool/net/demo/unix-subprocess.lua

**Comment by diyism** [comment]
Yes, with fork subprocess, I can see the "test" output on browser now.

But the redbean log still says "sudo: effective uid is not 0, is /usr/bin/sudo on a file system with the 'nosuid' option set or an NFS file system without root privileges?",

I can see somebody do this: "execve("/usr/bin/sudo",params, NULL);"  in stackoverflow(https://stackoverflow.com/a/27862437/264181)

Is there something different in the redbean envrionmnet with the normal environment?

My modified test.lua:
```
local unix = require "unix"
child = assert(unix.fork())

if child == 0 then
    cmd=assert(unix.commandv('/usr/bin/sudo'))
    unix.execve(cmd, {cmd, '/usr/bin/ls'})
end
Write('test')
```

**Comment by diyism** [comment]
It seems that the error ""sudo: effective uid is not 0, is /usr/bin/sudo on a file system with the 'nosuid' option set " often occurs in some container systems, is ".ape redbean.zip" like a container environment?

**Comment by diyism** [comment]
Cool, I found the Mako web server(https://github.com/RealTimeLogic/BAS), I've successfully runned my sudo command in it without inputing password:

$ wget makoserver.net/download/mako.linux-x64.tar.gz
$ tar xvf mako.linux-x64.tar.gz
$ ./mako -l::www
Mako Server. Version 3.8
......
Server listening on IPv4 port 9357

the file of www/index.lsp:
```
<html>
  <body>
        <?lsp response:write("Hello World<br/>")
        response:write("Hello Malcolm<br/>")
        ba.exec('/usr/bin/sudo /home/malcolm/lcus1.sh')
        ?>
    </h1>
  </body>
</html>

```

**Comment by diyism** [comment]
It seems that it has nothing to do with unix.execve(), even if the test.lua only has:
```
os.execute("sudo /home/malcolm/lcus1.sh")
Write('test2')
```
the redbean log will say: sudo: effective uid is not 0, is /usr/bin/sudo on a file system with the 'nosuid' option set or an NFS file system without root privileges?

**Comment by jart** [comment]
Ah I see what the issue is here. It's because we're calling `prctl(PR_SET_NO_NEW_PRIVS, 1, 0, 0, 0);` from our main() function in anticipation of using features like SECCOMP. We also drop Linux capabilities.

https://github.com/jart/cosmopolitan/blob/3665232d66c7af67399ce138208bedd1940d406c/tool/net/redbean.c#L7081-L7092

One of the reasons why we do it, is because we want redbean to be more secure by default. For example, some might consider it a feature rather than a bug that if redbean is compromised somehow, the attacker would not be able to execute commands as your root user. However there should be ideally a flag for disabling this. I'll reopen while I figure out the best way to do that.
